Source/SessionManager/SessionManager.swift

```diff
@@ -35,7 +35,8 @@ public typealias LaunchOptions = [UIApplication.LaunchOptionsKey : Any]
 }
 
 @objc public protocol SessionActivationObserver: class {
-    func sessionManagerActivated(userSession : ZMUserSession)
+    func sessionManagerWillChangeActiveUserSession(userSession: ZMUserSession?)
+    func sessionManagerChangedActiveUserSession(isDatabaseLocked: Bool)
```

Comment:
It's tricky to find good names for this situation for these two states that the active user session can be in. Maybe something like this? We **change** the active user session when we switch between the account and we **update** the current active session when it's either authenticated or when the database is locked/unlocked.

```suggestion
    func sessionManagerDidChangeActiveUserSession(_ userSession: ZMUserSession)
    func sessionManagerDidUpdateActiveUserSession(isDatabaseLocked: Bool)
```

Reply:
ping @johnxnguyen

Reply:
`sessionManagerDidUpdateActiveUserSession` is a little misleading because the session manager isn't doing anything to the active user session, it's just passing on information that the active user session database became locked / unlocked.

Perhaps something like

```swift
func activeUserSessionDatabaseLockDidChange(isLocked: Bool)
```

Reply:
yep, makes sense. Thanks a lot!!!

Reply:
I guess we added "sessionManager" at the beginning to be aligned with all the other methods of the delegate (see also `SessionManagerDelegate`) and because basically this call back is fired by the `SessionManager`.

Reply:
Anyway I will use you hint cause it is definitely more descriptive.

Reply:
Even if the session manager is just delegating information I would still write it from the session managers perspective, this makes the delegate methods consistent.

`sessionManagerDidObserveDatabaseLockChange(isLocked: Bool)`
`sessionManagerDidReportDatabaseLockChange(isLocked: Bool)`
`sessionManagerDidReportDatabaseLockChangeForActiveUserSession(isLocked: Bool)`

Reply:
ok at this point I go for 

`sessionManagerDidReportDatabaseLockChange(isLocked: Bool)`